numpy/core/src/umath/override.c

```diff
@@ -409,7 +409,8 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
             /* Check for sub-types to the right of obj. */
             for (j = i + 1; j < noa; j++) {
                 other_obj = with_override[j];
-                if (PyObject_Type(other_obj) != PyObject_Type(obj) &&
+                if (other_obj != NULL &&
+                    PyObject_Type(other_obj) != PyObject_Type(obj) &&
```

Comment:
Should have an extra indent here to clearly separate the boolean expression from the body of the if.